src/Umbraco.Web.UI.Client/src/utility.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * A friendly utility collection to replace AngularJs' ng-functions
+ * If it doesn't exist here, it's probably available as vanilla JS
+ * 
+ * Still carries a dependency on underscore, but if usages of underscore from 
+ * elsewhere in the codebase can instead use these methods, the underscore
+ * dependency will be nicely abstracted and can be removed/swapped later
+ * 
+ * This collection is open to extension...
+ */
+(function (window) {
+
+    /**
+     * Equivalent to angular.noop
+     */
+    const noop = () => { };
+
+    /**
+     * Equivalent to angular.copy
+     * Abstraction of underscore.clone
+     */
+    const copy = val => _.clone(val);
```

Comment:
@nathanwoulfe `angular.copy` does a deep clone, while `_.clone` only does a shallow clone. From what I can tell, underscore doesn't have a deep clone function, so we'll most likely have to build this our selves. Or as Angular.js is MIT, we may borrow it from their source code (they seem to cover a lot of different cases, and it's probably well tested).

Exhibit A:

```javascript
var a = { hello: { world: true } };

var b = angular.copy(a);

a.hello.world = false;

console.log(b.hello.world); // outputs "true"
```

Exhibit B:

```javascript
var a = { hello: { world: true } };

var b = _.clone(a);

a.hello.world = false;

console.log(b.hello.world); // outputs "false"
```

Reply:
@abjerner I think the most robust way would be to include the AngularJs version - removes the risk of things breaking due to differing implementations. That said, that's a complex piece of code, so wondering if instead, for now at least, Utilities.copy() is simply a facade to angular.copy() - would serve the same purpose of isolating references, and can be dealt with in due course without blocking other changes. Same for angular.equal.

We could look at importing individual functions from lodash (where cloneDeep is a thing), but that means changes to the gulp build to introduce browserify or similar to manage getting the CommonJs stuff into the browser...

Reply:
It's not like we're skipping Angular for Umbraco 8 anytime soon, so I agree that having `Utitilities.copy` call `angular.copy()` for now would be an acceptable solution. Adding another dependency should be avoided if possible.